test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -90,11 +90,12 @@ public void setup() {
         assertTrue(segmentStoreInstance.isRunning());
         log.info("Pravega Segmentstore service instance details: {}", segmentStoreInstance.getServiceDetails());
 
-        //executor service
-        executorService = Executors.newScheduledThreadPool(NUM_READERS + NUM_WRITERS);
+        //num. of readers + num. of writers + 1 to run checkScale operation
+        executorService = Executors.newScheduledThreadPool(NUM_READERS + NUM_WRITERS + 1);
         //get Controller Uri
         controller = new ControllerImpl(controllerURIDirect);
         testState = new TestState();
+        testState.writersListComplete.set(0, testState.writersComplete);
```

Comment:
We can't do this, `set` is used only to replace, so we need to `add` rather than `set`as you're doing before. We don't need the `set`, though.